openai/azure-openai/runtime/src/main/java/io/quarkiverse/langchain4j/azure/openai/AzureOpenAiImageModel.java

```diff
@@ -0,0 +1,263 @@
+package io.quarkiverse.langchain4j.azure.openai;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.model.openai.OpenAiModelName.DALL_E_2;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import dev.ai4j.openai4j.image.GenerateImagesRequest;
+import dev.ai4j.openai4j.image.GenerateImagesResponse;
+import dev.langchain4j.data.image.Image;
+import dev.langchain4j.model.image.ImageModel;
+import dev.langchain4j.model.output.Response;
+import io.quarkiverse.langchain4j.openai.QuarkusOpenAiClient;
+
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class AzureOpenAiImageModel implements ImageModel {
+    private final String modelName;
+    private final String size;
+    private final String quality;
+    private final String style;
+    private final Optional<String> user;
+    private final String responseFormat;
+    private final Integer maxRetries;
+    private final Optional<Path> persistDirectory;
+
+    private final QuarkusOpenAiClient client;
+
+    public AzureOpenAiImageModel(String endpoint, String apiKey, String apiVersion, String modelName, String size,
+            String quality, String style, Optional<String> user, String responseFormat, Duration timeout,
+            Integer maxRetries, Boolean logRequests, Boolean logResponses,
+            Optional<Path> persistDirectory) {
+        this.modelName = modelName;
+        this.size = size;
+        this.quality = quality;
+        this.style = style;
+        this.user = user;
+        this.responseFormat = responseFormat;
+        this.maxRetries = maxRetries;
+        this.persistDirectory = persistDirectory;
+
+        this.client = QuarkusOpenAiClient.builder()
+                .baseUrl(ensureNotBlank(endpoint, "endpoint"))
+                .azureApiKey(apiKey)
+                .apiVersion(apiVersion)
+                .callTimeout(timeout)
+                .connectTimeout(timeout)
+                .readTimeout(timeout)
+                .writeTimeout(timeout)
+                .logRequests(logRequests)
+                .logResponses(logResponses)
+                .build();
+    }
+
+    @Override
+    public Response<Image> generate(String prompt) {
+        GenerateImagesRequest request = requestBuilder(prompt).build();
+
+        GenerateImagesResponse response = withRetry(() -> client.imagesGeneration(request), maxRetries).execute();
```

Comment:
No lambdas please

Reply:
This was copy pasted from your own code in the openai-vanilla module.....

Reply:
Which should be fixed too then 😉

Reply:
I'll leave that for another day and another PR. Want to keep this one focused on the task at hand. :)

Reply:
Right, I wasn't proposing to do it now

Reply:
Done